### PR TITLE
clippy 1.65

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: crates/compiler/can/src/abilities.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ pub struct MemberVariables {
 
 /// The member and its signature is defined locally, in the module the store is created for.
 /// We need to instantiate and introduce this during solving.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
```
Comment (ayazhafiz) [inline crates/compiler/can/src/abilities.rs:31]: Did clippy catch this?

File: crates/compiler/can/src/expr.rs [hunk 2]
```diff
@@ -1652,7 +1651,7 @@ fn canonicalize_var_lookup(
 }
 
 /// Currently uses the heuristic of "only inline if it's a builtin"
-pub fn inline_calls(var_store: &mut VarStore, scope: &mut Scope, expr: Expr) -> Expr {
+pub fn inline_calls(var_store: &mut VarStore, expr: Expr) -> Expr {
```
Comment (ayazhafiz) [inline crates/compiler/can/src/expr.rs:1654]: What a great catch, super nice that they made this a warning by default now

File: crates/compiler/can/src/abilities.rs (hunk 1 above)
Comment (folkertdev) [reply crates/compiler/can/src/abilities.rs:31]: yes it complains now if `PartialEq` is derived, and `Eq` can be but isn't

File: crates/compiler/can/src/expr.rs (hunk 2 above)
Comment (folkertdev) [reply crates/compiler/can/src/expr.rs:1654]: yes it's amazing

File: crates/compiler/can/src/abilities.rs (hunk 1 above)
Comment (ayazhafiz) [reply crates/compiler/can/src/abilities.rs:31]: wow@

File: crates/compiler/can/src/expr.rs (hunk 2 above)
Comment (rtfeldman) [reply crates/compiler/can/src/expr.rs:1654]: Wow, "argument is only used in recursive calls" I guess? That's awesome - we should do that! 😍

**Review by rtfeldman — APPROVED** [review]
Very nice, thank you @folkertdev! 🎉

**Pull request merged**